x-pack/plugins/maps/public/classes/styles/vector/components/vector_style_editor.tsx

```diff
@@ -7,34 +7,95 @@
 import _ from 'lodash';
 import React, { Component, Fragment } from 'react';
 
+import { i18n } from '@kbn/i18n';
+import { EuiSpacer, EuiButtonGroup, EuiFormRow, EuiSwitch, EuiSwitchEvent } from '@elastic/eui';
 import { VectorStyleColorEditor } from './color/vector_style_color_editor';
 import { VectorStyleSizeEditor } from './size/vector_style_size_editor';
+// @ts-expect-error
 import { VectorStyleSymbolizeAsEditor } from './symbol/vector_style_symbolize_as_editor';
 import { VectorStyleIconEditor } from './symbol/vector_style_icon_editor';
 import { VectorStyleLabelEditor } from './label/vector_style_label_editor';
+// @ts-expect-error
 import { VectorStyleLabelBorderSizeEditor } from './label/vector_style_label_border_size_editor';
+// @ts-expect-error
 import { OrientationEditor } from './orientation/orientation_editor';
 import { getDefaultDynamicProperties, getDefaultStaticProperties } from '../vector_style_defaults';
 import { DEFAULT_FILL_COLORS, DEFAULT_LINE_COLORS } from '../../color_palettes';
-import { i18n } from '@kbn/i18n';
 
-import { EuiSpacer, EuiButtonGroup, EuiFormRow, EuiSwitch } from '@elastic/eui';
 import {
   LABEL_BORDER_SIZES,
   VECTOR_STYLES,
   STYLE_TYPE,
   VECTOR_SHAPE_TYPE,
 } from '../../../../../common/constants';
-import { createStyleFieldsHelper } from '../style_fields_helper';
-
-export class VectorStyleEditor extends Component {
-  state = {
-    styleFields: [],
-    defaultDynamicProperties: getDefaultDynamicProperties(),
-    defaultStaticProperties: getDefaultStaticProperties(),
-    supportedFeatures: undefined,
-    selectedFeature: null,
-  };
+import { createStyleFieldsHelper, StyleField, StyleFieldsHelper } from '../style_fields_helper';
+import {
+  ColorDynamicOptions,
+  ColorStaticOptions,
+  DynamicStylePropertyOptions,
+  IconDynamicOptions,
+  IconStaticOptions,
+  LabelDynamicOptions,
+  LabelStaticOptions,
+  SizeDynamicOptions,
+  SizeStaticOptions,
+  StaticStylePropertyOptions,
+  StylePropertyOptions,
+  VectorStylePropertiesDescriptor,
+} from '../../../../../common/descriptor_types';
+import { IStyleProperty } from '../properties/style_property';
+import { SymbolizeAsProperty } from '../properties/symbolize_as_property';
+import { LabelBorderSizeProperty } from '../properties/label_border_size_property';
+import { StaticTextProperty } from '../properties/static_text_property';
+import { StaticSizeProperty } from '../properties/static_size_property';
+import { IVectorLayer } from '../../../layers/vector_layer/vector_layer';
+
+export interface StyleProperties {
+  [key: string]: IStyleProperty<StylePropertyOptions>;
```

Comment:
change from `string` to `VECTOR_STYLES` ?

Reply:
That causes all kinds of typescripting hell that I could not figure out. I left it as string so my head did not explode.